editor/inserter/menu.js

```diff
@@ -92,25 +98,39 @@ class InserterMenu extends Component {
 		)( blockTypes );
 	}
 
-	findByIncrement( blockTypes, increment = 1 ) {
-		// Prepend a fake search block to the list to cycle through.
-		const list = [ { name: 'search' }, ...blockTypes ];
+	getVisibleBlocksForTab( blockTypes ) {
+		if ( this.state.tab === 'embeds' ) {
+			return blockTypes.filter( ( block ) => block.category === 'embed' );
+		}
+
+		if ( this.state.tab === 'blocks' ) {
+			return blockTypes.filter( ( block ) => block.category !== 'embed' );
+		}
 
-		const currentIndex = findIndex( list, ( blockType ) => this.state.currentFocus === blockType.name );
+		return blockTypes.filter( ( block ) => block.category === 'common' );
+	}
+
+	findByIncrement( blockTypes, increment = 1 ) {
+		const currentIndex = findIndex( blockTypes, ( blockType ) => this.state.currentFocus === blockType.name );
 		const nextIndex = currentIndex + increment;
-		const highestIndex = list.length - 1;
+		const highestIndex = blockTypes.length - 1;
 		const lowestIndex = 0;
 
 		if ( nextIndex > highestIndex ) {
-			return list[ lowestIndex ].name;
+			return 'search';
 		}
 
 		if ( nextIndex < lowestIndex ) {
-			return list[ highestIndex ].name;
+			return 'search';
 		}
 
 		// Return the name of the next block type.
-		return list[ nextIndex ].name;
+		const block = blockTypes[ nextIndex ];
```

Comment:
Can we call this variable `blockType` to avoid confusion? Also the argument and name of `isDisabledBlockType`.